File: Producto/frontend/src/remote/DTO/HojaCompraItemDTO.js
/**
 * DTO para representar un ítem de Hoja de Compra.
 * Espejo de HojaCompraItemDTO.java del backend.
 */
export class HojaCompraItemDTO {
    constructor(data = {}) {
        this.idHCItem = data.idHCItem ?? null;
        this.hcId = data.hcId ?? null;
        this.tipoInsumo = data.tipoInsumo ?? '';
        this.insumoId = data.insumoId ?? data.articuloId ?? null;
        this.articuloId = data.articuloId ?? data.insumoId ?? null;
        this.nombreInsumo = data.nombreInsumo ?? '';
        this.consumoUnitario = data.consumoUnitario ?? 0;
        this.cantidadOP = data.cantidadOP ?? 0;
        this.cantidadRequerida = data.cantidadRequerida ?? 0;
        this.precioUnitarioRef = data.precioUnitarioRef ?? 0;
        this.cantidadStock = data.cantidadStock ?? 0;
        this.cantidadAComprar = data.cantidadAComprar ?? null;
        this.modificado = data.modificado ?? false;
        this.justificacionModificacion = data.justificacionModificacion ?? '';
        this.proveedorId = data.proveedorId ?? null;
        this.proveedorNombre = data.proveedorNombre ?? '';
        this.ocId = data.ocId ?? null;
        this.numeroOC = data.numeroOC ?? '';
    }
}

File: Producto/backend/src/main/java/backend/com/produccion/application/dto/HojaCompraItemDTO.java
package backend.com.produccion.application.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class HojaCompraItemDTO {

    private Long idHCItem;
    private Long hcId;
    private String tipoInsumo;
    private Integer articuloId;
    private String nombreInsumo;
    private BigDecimal consumoUnitario;
    private Integer cantidadOP;
    private BigDecimal cantidadRequerida;
    private BigDecimal precioUnitarioRef;
    
    private BigDecimal cantidadStock;
    private BigDecimal cantidadAComprar;
    private Boolean modificado;
    private String justificacionModificacion;
    
    // Nuevos campos para proveedor y OC vinculada
    private Long proveedorId;
    private String proveedorNombre;
    private Long ocId;
    private String numeroOC;

    /** false si el ítem fue agregado manualmente ("no presupuestado"). */
    private Boolean presupuestado;
}
